api/v1alpha1/key_types.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2019 Christopher Hein <me@chrishein.com>.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KeySpec defines the desired state of Key
+type KeySpec struct {
+	// +optional
+	// ReadOnly determines whether the key has write access to the repository
+	ReadOnly bool `json:"readOnly"`
+
+	// +kubebuilder:validation:MaxLength 253
+	// RepositoryRef points to a Repository in the same Namespace that the Key is for
+	RepositoryRef string `json:"repositoryRef"`
+
+	// +optional
+	// SecretTemplate sets annotations and labels on the resulting Secret of this Key
+	SecretTemplate KeySecretTemplate `json:"secretTemplate,omitempty"`
+}
+
+// KeySecretTemplate is a template for creating Secrets that hold Key data
+type KeySecretTemplate struct {
+	// Map of string keys and values that can be used to organize and categorize
+	// (scope and select) objects. May match selectors of replication controllers
+	// and services.
+	// More info: http://kubernetes.io/docs/user-guide/labels
+	// +optional
+	Labels map[string]string `json:"labels,omitempty" protobuf:"bytes,11,rep,name=labels"`
+
+	// Annotations is an unstructured key value map stored with a resource that may be
+	// set by external tools to store and retrieve arbitrary metadata. They are not
+	// queryable and should be preserved when modifying objects.
+	// More info: http://kubernetes.io/docs/user-guide/annotations
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,12,rep,name=annotations"`
```

Comment:
We probably don't need the `protobuf` tags but it's fine to leave in.

Reply:
Yep! Just copied over verbatim from the upstream versions